### PR TITLE
Alamin/show available graph

### PR DESCRIPTION
**Pull request by Full-stack-Web-Dev-BD**
Here is the PR 

## Summary by CodeRabbit

- **New Features**
	- Introduced a `Combobox` component for enhanced dropdown selections, including adding new options and searching.
	- Implemented a `GraphsList` component for selecting graphs via Cypher queries.
	- Developed components for creating accessible accordions (`Accordion`), command interfaces (`Command`), customizable dialog boxes (`Dialog`), and popovers (`Popover`).
	- Added a `Separator` component for visually separating content within the UI.

- **Enhancements**
	- Improved data fetching and state management in the `Page` and `Query` components.

- **Documentation**
	- Documented the purpose and usage of newly introduced UI components and functionalities.

### CODE REVIEW

File: yarn.lock
Comment (gkorland) [inline yarn.lock:1]: please remove the yarn.lock from the PR

File: app/graph/Example.tsx
Comment (gkorland) [inline app/graph/Example.tsx:1]: Why do you need Example.tsx?

File: app/graph/page.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { toast } from "@/components/ui/use-toast";
 import CytoscapeComponent from 'react-cytoscapejs'
 import cytoscape from 'cytoscape';
 import fcose from 'cytoscape-fcose';
-import { useRef, useState } from "react";
+import { useEffect, useRef, useState } from "react";
```
Comment (gkorland) [inline app/graph/page.tsx:7]: Where do you use useEffect in the code?

File: components/ui/accordion.tsx
Comment (gkorland) [inline components/ui/accordion.tsx:1]: where do you use Accordion.tsx? why did you add it?

**Review by gkorland — CHANGES_REQUESTED** [review]

File: app/components/combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                  <CommandItem>Create new {entityType}...</CommandItem>
+                </DialogTrigger>
+                <DialogContent>
+                  <DialogHeader>
+                    <DialogTitle>Create a new {entityType}?</DialogTitle>
+                    <DialogDescription>
+                      <Input type="text" ref={inputRef} id="create" name="create" onKeyDown={handleKeyDown} placeholder={`${entityType} name ...`} />
+                    </DialogDescription>
+                  </DialogHeader>
+                  <Button className="bg-blue-600 p-4 text-slate-50" type="submit" onClick={onAddOption}>Create</Button>
```
Comment (gkorland) [inline app/components/combobox.tsx:107]:
```suggestion
                  <Button className="p-4" type="submit" onClick={onAddOption}>Create</Button>
```

File: app/graph/page.tsx (hunk 1 above)
Comment (gkorland) [inline app/graph/page.tsx:7]:
```suggestion
import { useRef, useState } from "react";
```

File: app/graph/GraphList.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }, [toast])
+
+    function setOptions(newGraphs: SetStateAction<string[]>) {
+        setGraphs(newGraphs)
+        setSelectedValue(graphs[graphs.length - 1])
+    }
+
+    function addOption(newGraphs: string) {
+        graphs.push(newGraphs)
+        setOptions(graphs)
```
[the remark this answers is not in the document]
Comment (Full-stack-Web-Dev-BD) [reply app/graph/GraphList.tsx:39]: It resolved  with :
`    function setOptions(newGraphs: SetStateAction<string[]>) {
        setGraphs((prevGraphs:any) => [...prevGraphs, newGraphs]);
        setSelectedValue(newGraphs[newGraphs.length - 1]);
    }
`
Thank you

File: app/graph/query.tsx [hunk 4]
```diff
@@ -27,20 +30,24 @@ export function Query(params: {
         setQuery(event.target.value)
     }
 
-    function updateGraph(event: React.ChangeEvent<HTMLInputElement>) {
-        setGraphName(event.target.value)
-    }
-
+    // function updateGraph(event: React.ChangeEvent<HTMLInputElement>) {
+    //     setGraphName(event.target.value)
+    // }
```
[the remark this answers is not in the document]
Comment (Full-stack-Web-Dev-BD) [reply app/graph/query.tsx:35]: Removed :)

File: yarn.lock
Comment (Full-stack-Web-Dev-BD) [reply yarn.lock:1]: Removed  from my Fork , It  might be generated durinin installation of  dependency :)

File: app/graph/Example.tsx
Comment (Full-stack-Web-Dev-BD) [reply app/graph/Example.tsx:1]: It was needed  to show  example graph , as we are not  showing  any example  , currently  we dont need it -Removed

File: app/graph/page.tsx (hunk 1 above)
Comment (Full-stack-Web-Dev-BD) [reply app/graph/page.tsx:7]: I may have imported it for testing purposes, and now it needs to be removed-Done

File: components/ui/accordion.tsx
Comment (Full-stack-Web-Dev-BD) [reply components/ui/accordion.tsx:1]: there is no use of this - Removed

File: app/components/combobox.tsx (hunk 2 above)
Comment (Full-stack-Web-Dev-BD) [reply app/components/combobox.tsx:107]: Updated  - Done

File: app/graph/page.tsx (hunk 1 above)
Comment (Full-stack-Web-Dev-BD) [reply app/graph/page.tsx:7]: Its already Done

**Comment by gkorland** [comment]
@Full-stack-Web-Dev-BD please notice the CI error:
```
npm ERR! code EUSAGE
npm ERR! 
npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
```

**Pull request merged**